engine/runtime/src/main/java/org/enso/interpreter/node/typecheck/LazyCheckRootNode.java

```diff
@@ -0,0 +1,46 @@
+package org.enso.interpreter.node.typecheck;
+
+import com.oracle.truffle.api.TruffleLanguage;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.nodes.RootNode;
+import org.enso.interpreter.node.BaseNode;
+import org.enso.interpreter.node.callable.thunk.ThunkExecutorNode;
+import org.enso.interpreter.runtime.callable.argument.ArgumentDefinition;
+import org.enso.interpreter.runtime.callable.function.Function;
+import org.enso.interpreter.runtime.callable.function.FunctionSchema;
+
+final class LazyCheckRootNode extends RootNode {
```

Comment:
Why this class does not extend `EnsoRootNode`? By extending it, we would have functionality like source section and associated scopes.

Reply:
Probably because it represents _internal function only_:
```java
  Function wrapThunk(Function thunk) {
    return new Function(getCallTarget(), thunk.getScope(), SCHEMA, new Object[] {thunk}, null);
  }
```
In any case, I am not aware of any missing source sections or scopes.